src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -9,23 +9,52 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class BoundStatement : IOperation
+    internal partial class BoundNode : IOperation, IOperationWithChildren
```

Comment:
Is there an advantage to having a separate interface for `IOperationWithChildren`? Aren't callers forced to check for an implementation of `IOperationWithChildren`, then check `.Children.IsEmpty` or perhaps even `.IsDefault`?